packages/components/src/internal/components/entities/models.ts

```diff
@@ -507,7 +507,17 @@ export class EntityIdCreationModel extends Record({
         const rows = editorModel
             .getRawData(queryGridModel, false)
             .valueSeq()
-            .reduce((rows, row) => rows.push(row.toMap()), List<Map<string, any>>());
+            .reduce((rows, row) => {
+                let map = row.toMap();
+                if (extraColumnsToInclude && extraColumnsToInclude.length > 0)
```

Comment:
Could get rid of the length check here since the for loop wouldn't do any harm if there's nothing to iterate on.